x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -376,6 +376,56 @@ export function jobRoutes({ xpackMainPlugin, router }: RouteInitialization) {
     })
   );
 
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/:jobId/results/buckets  Obtain bucket scores for the specified job ID
+   * @apiName GetOverallBuckets
+   * @apiDescription The get buckets API presents a chronological view of the records, grouped by bucket.
+   *
+   * @apiParam {String} jobId Job ID.
+   * @apiParam {Number} timestamp.
+   *
+   * @apiSuccess {Number} count
+   * @apiSuccess {Object[]} buckets
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/results/buckets/{timestamp?}',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+          timestamp: schema.maybe(schema.string()),
+        }),
+        body: schema.object({
+          anomaly_score: schema.maybe(schema.number()),
+          desc: schema.maybe(schema.boolean()),
+          end: schema.maybe(schema.string()),
+          exclude_interim: schema.maybe(schema.boolean()),
+          expand: schema.maybe(schema.boolean()),
+          'page.from': schema.maybe(schema.number()),
+          'page.size': schema.maybe(schema.number()),
+          sort: schema.maybe(schema.string()),
+          start: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.buckets', {
+          jobId: request.params.jobId,
+          timestamp: request.params.timestamp,
+          ...request.body,
+        });
+        return response.ok({
+          body: { ...results },
```

Comment:
nit: there is no need in spread operator

Reply:
good point. a lot of our endpoints are like this. i'll change them all in this PR.